utils.py

```diff
@@ -3,10 +3,14 @@
 import warnings
 import matplotlib.pyplot as plt
 import numpy as np
+from PIL import Image
```

Comment:
I think PIL.Image is not gonna be used right?

Reply:
right, I added some lines to plot some examples during the running time, and then I removed them.. 
So yea, I will remove it.